src/audiomixerboard.cpp

```diff
@@ -1310,6 +1313,167 @@ void CAudioMixerBoard::SetAllFaderLevelsToNewClientLevel()
+            if ( group == INVALID_INDEX )
+            {
+                group = MAX_NUM_FADER_GROUPS;
+            }
+
+            if ( leveldB >= AUTO_FADER_NOISE_THRESHOLD_DB )
+            {
+                vecMaxLevel[group] = fmax ( vecMaxLevel[group], leveldB );
+                levels[group].Add ( leveldB );
+            }
+            ++vecChannelsPerGroup[group];
+        }
+    }
+
+    // sort levels for later median computation
+    for ( int i = 0; i < MAX_NUM_FADER_GROUPS + 1; ++i )
+    {
+        std::sort ( levels[i].begin(), levels[i].end() );
+    }
+
+    // compute the number of active groups (at least one channel)
+    int cntActiveGroups = 0;
+    for ( int i = 0; i < MAX_NUM_FADER_GROUPS; ++i )
+    {
+        cntActiveGroups += vecChannelsPerGroup[i] > 0;
+    }
+
+    // only my channel is active, nothing to do
+    if ( cntActiveGroups == 0 &&
+        vecChannelsPerGroup[MAX_NUM_FADER_GROUPS] == 0 )
+    {
+        return;
+    }
+
+    // compute target level for each group
+    // (prevent clipping when each group contributes at maximum level)
+    float targetLevelPerGroup = -20.0f * log10 (
+        std::max ( cntActiveGroups, 1 ) );
+
+    // compute target levels for the channels of each group individually
+    float vecTargetChannelLevel[MAX_NUM_FADER_GROUPS + 1];
+    float levelOffset = 0.0f;
+    float minFader = 0.0f;
+    for ( int i = 0; i < MAX_NUM_FADER_GROUPS + 1; ++i )
+    {
+        // compute the target level for each channel in the current group
+        // (prevent clipping when each channel in this group contributes at
+        // the maximum level)
+        vecTargetChannelLevel[i] = vecChannelsPerGroup[i] > 0 ?
+            targetLevelPerGroup - 20.0f * log10 ( vecChannelsPerGroup[i] ) :
+            0.0f;
+
+        // get median level
+        int cntChannels = levels[i].Size();
+        if ( cntChannels == 0 )
+        {
+            continue;
+        }
+        float refLevel = levels[i][cntChannels / 2];
+
+        // since we can only attenuate channels but not amplify, we have to
+        // check that the reference channel can be brought to the target
+        // level
+        if ( refLevel < vecTargetChannelLevel[i] )
+        {
+            // otherwise, we adjust the level offset in such a way that
+            // the level can be reached
+            levelOffset = fmin ( levelOffset,
+                refLevel - vecTargetChannelLevel[i] );
+
+            // compute the minimum necessary fader setting
+            minFader = fmin ( minFader, -vecMaxLevel[i] +
+                vecTargetChannelLevel[i] + levelOffset );
+        }
+    }
+
+    // take minimum fader value into account
+    // very weak channels would actually require strong channels to be
+    // attenuated to a large amount; however, the attenuation is limited by
+    // the faders
+    if ( minFader < -AUD_MIX_FADER_RANGE_DB )
+        levelOffset += -AUD_MIX_FADER_RANGE_DB - minFader;
```

Comment:
Not sure about the style here. Would we usually use brackets here?

Reply:
Ok, I will fix that with the next commit.